<h1>Содержание<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Описание" data-toc-modified-id="Описание-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Описание</a></span></li><li><span><a href="#Импорт-библиотек-и-данных" data-toc-modified-id="Импорт-библиотек-и-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Импорт библиотек и данных</a></span></li><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Предобработка данных</a></span></li><li><span><a href="#Расчеты-RMF" data-toc-modified-id="Расчеты-RMF-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Расчеты RMF</a></span><ul class="toc-item"><li><span><a href="#Frequency" data-toc-modified-id="Frequency-4.1"><span class="toc-item-num">4.1&nbsp;&nbsp;</span>Frequency</a></span></li><li><span><a href="#Monetary" data-toc-modified-id="Monetary-4.2"><span class="toc-item-num">4.2&nbsp;&nbsp;</span>Monetary</a></span></li><li><span><a href="#Recency" data-toc-modified-id="Recency-4.3"><span class="toc-item-num">4.3&nbsp;&nbsp;</span>Recency</a></span></li><li><span><a href="#Объдинение-таблиц" data-toc-modified-id="Объдинение-таблиц-4.4"><span class="toc-item-num">4.4&nbsp;&nbsp;</span>Объдинение таблиц</a></span></li><li><span><a href="#Расчет-степеней-R,-M,-F" data-toc-modified-id="Расчет-степеней-R,-M,-F-4.5"><span class="toc-item-num">4.5&nbsp;&nbsp;</span>Расчет степеней R, M, F</a></span></li></ul></li><li><span><a href="#Выделение-сегментов" data-toc-modified-id="Выделение-сегментов-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Выделение сегментов</a></span><ul class="toc-item"><li><span><a href="#Сегмент-attention" data-toc-modified-id="Сегмент-attention-5.1"><span class="toc-item-num">5.1&nbsp;&nbsp;</span>Сегмент attention</a></span></li><li><span><a href="#Сегмент-лидеров" data-toc-modified-id="Сегмент-лидеров-5.2"><span class="toc-item-num">5.2&nbsp;&nbsp;</span>Сегмент лидеров</a></span></li><li><span><a href="#Сегмент-потенциала" data-toc-modified-id="Сегмент-потенциала-5.3"><span class="toc-item-num">5.3&nbsp;&nbsp;</span>Сегмент потенциала</a></span></li></ul></li><li><span><a href="#Подготовка-презентации" data-toc-modified-id="Подготовка-презентации-6"><span class="toc-item-num">6&nbsp;&nbsp;</span>Подготовка презентации</a></span></li><li><span><a href="#Вывод" data-toc-modified-id="Вывод-7"><span class="toc-item-num">7&nbsp;&nbsp;</span>Вывод</a></span></li></ul></div>

# Портрет пользователя

## Описание

**Классификация пользователей по методологии RFM**

Методология RFM вполне простая: исследователю нужно присвоить каждому заказчику степень «1», «2», «3» в зависимости от:
- общей суммы отгруженных заказов (предоставленных и оплаченных услуг) за весь период работы с контрагентом. За нее отвечает показатель Monetary (M)

- частоты заказов – Frequency (F)

- насколько недавно была совершена последняя сделка – Recency (R)

R=3, F=1, M=1 = заказчики с самым высоким LTV и высокой частотой заказов, которые делали последний заказ давно, находятся в оттоке. Такие контрагенты требуют экстренных мероприятий по выявлению причин оттока и попыток вернуть клиента

M=1, F=1, R=1 = потенциально опинион-лидеры, адвокаты вашего бренда. Также полезно изучить качественный состав таких контрагентов при большом количестве сегментов целевой аудитории в вашей базе

M=2/3, F=1, R=1= в таком сегменте здорово проводить мероприятия, направленные на увеличение среднего чека

RFM-анализ местами перекликается с известным Законом Парето, гласящим, что 80% результатов происходят благодаря 20% усилий. Когда на этот закон смотрят через призму маркетинга, это означает, что 80% всех ваших продаж исходят от 20% наиболее лояльных и постоянных клиентов. 

![Закон Парето](https://leonardo.osnova.io/e5931522-b3ac-54f0-9f10-52f125c1dd24/-/preview/1800/-/format/webp/)

**Задача**

Выделить по покупкам за этот год сегменты пользовавателей для анализа продактом, поставновки задачи для клинтского сервиса по работе с каждым сегментом. Рассматриваемый перод - текущий год (01.01.2023 - 25.09.2023)

## Импорт библиотек и данных

In [1]:
# Импорт библиотек
import pandas as pd
import numpy as np
import datetime
import matplotlib.pyplot as plt
import math
import warnings

# отключение предупреждений об ошибках
warnings.filterwarnings('ignore')

Импортируем таблицу полисов с дашборда DataLens, период покупок 01.01.2023 - 25.09.2023

In [2]:
data = pd.read_csv('DataExport_2023-09-25_10-51-32.csv', sep=';', on_bad_lines='skip')
data.head()

policy_id  payment_id source     user_id                 date company  \
0      62013        8287    web  7000017232  24.09.2023 14:37:33   Zetta   
1      62008        8413    app  7000013914  24.09.2023 13:10:13   Zetta   
2      62001        8110    web  7000017006  24.09.2023 09:52:50   Zetta   
3      62000        8395    app  7000014990  24.09.2023 09:49:22     ВСК   
4      61999        8391    web  7000017076  24.09.2023 09:09:50   Zetta   

      type  number_policy  cost_of_policy  payment  duration_hours   km  \
0  parking              1          899.78   899.78             720    0   
1    theft              2         1266.32  1266.32            2184    0   
2       km              1          828.75   828.75            8784  500   
3  parking             11           58.74    58.74               1    0   
4    combo              1         2806.40  2806.40             168    0   

   odometer promocode        surname                          vehicle  \
0       NaN       NaN         Ченина                     Kia Rio 2014   
1       NaN       NaN  Скриголовская                  Opel Astra 2013   
2       0.0       NaN      Чернецкий               Chery Tiggo 8 2020   
3       NaN       NaN        Новиков  Land Rover Discovery Sport 2020   
4       NaN       NaN      Федоркова           Volkswagen Tiguan 2012   

  user_source  
0         vsp  
1    sber_web  
2         vsp  
3         app  
4    sber_web

## Предобработка данных

Отобразим информацю о датафрейме

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 753 entries, 0 to 752
Data columns (total 17 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   policy_id       753 non-null    int64  
 1   payment_id      753 non-null    int64  
 2   source          753 non-null    object 
 3   user_id         753 non-null    int64  
 4   date            753 non-null    object 
 5   company         753 non-null    object 
 6   type            753 non-null    object 
 7   number_policy   753 non-null    int64  
 8   cost_of_policy  753 non-null    float64
 9   payment         753 non-null    float64
 10  duration_hours  753 non-null    int64  
 11  km              753 non-null    int64  
 12  odometer        147 non-null    float64
 13  promocode       93 non-null     object 
 14  surname         753 non-null    object 
 15  vehicle         743 non-null    object 
 16  user_source     753 non-null    object 
dtypes: float64(3), int64(6), object(8)


Всего 753 полиса, пропуски есть только в необязательных полях: odometer с показаниями одометра, когда покупаются километры, promocode применение промокода, vehicle авто

Преобразуем типы данных к нужным:

In [4]:
# преобразование даты покупки
data['date'] = pd.to_datetime(data['date'], format='%d.%m.%Y %H:%M:%S')
data['date']

0     2023-09-24 14:37:33
1     2023-09-24 13:10:13
2     2023-09-24 09:52:50
3     2023-09-24 09:49:22
4     2023-09-24 09:09:50
              ...        
748   2023-01-06 04:09:08
749   2023-01-05 08:00:44
750   2023-01-05 04:39:23
751   2023-01-03 16:41:20
752   2023-01-03 09:11:13
Name: date, Length: 753, dtype: datetime64[ns]

Проверим наличие строк дубликатов:

In [5]:
data.duplicated().sum()

0

## Расчеты RMF

### Frequency

Соберем частоту закзазов по пользователю: Frequency 

In [6]:
frequency = data.groupby('user_id').agg({'policy_id': 'count'}).reset_index().rename(
    columns={'policy_id': 'number_policies'})
frequency

user_id  number_policies
0    7000000090                1
1    7000000477                2
2    7000000549                2
3    7000001437                1
4    7000001601               25
..          ...              ...
378  7000017257                1
379  7000017261                1
380  7000017263                2
381  7000017284                1
382  7000017301                1

[383 rows x 2 columns]

Выведем распределение частоты методом describe

In [7]:
frequency.describe(percentiles = [.2, .4, .6, .8, .9, .99]).round()

user_id  number_policies
count  3.830000e+02            383.0
mean   7.000014e+09              2.0
std    3.321000e+03              3.0
min    7.000000e+09              1.0
20%    7.000012e+09              1.0
40%    7.000015e+09              1.0
50%    7.000015e+09              1.0
60%    7.000016e+09              1.0
80%    7.000017e+09              2.0
90%    7.000017e+09              4.0
99%    7.000017e+09             14.0
max    7.000017e+09             39.0

- 80% пользователей совершали покупку 2 и менее раз
- 90% пользователей совершали покупку 4 и менее раз
- количество покупок на пользователя от 1 до 39

### Monetary

Найдем общую сумму заказов по каждому пользователю (LTV по общему доходу):

In [8]:
monetary = data.groupby('user_id').agg({'cost_of_policy': 'sum'}).reset_index().rename(
    columns={'cost_of_policy': 'amount'})
monetary

user_id   amount
0    7000000090   107.85
1    7000000477   295.50
2    7000000549  1134.26
3    7000001437   314.00
4    7000001601   956.55
..          ...      ...
378  7000017257  1008.75
379  7000017261   971.70
380  7000017263  1381.99
381  7000017284  1327.89
382  7000017301   491.64

[383 rows x 2 columns]

In [9]:
monetary.describe(percentiles = [.2, .4, .6, .8, .9, .99]).round()

user_id   amount
count  3.830000e+02    383.0
mean   7.000014e+09   1790.0
std    3.321000e+03   2093.0
min    7.000000e+09      5.0
20%    7.000012e+09    475.0
40%    7.000015e+09    888.0
50%    7.000015e+09   1083.0
60%    7.000016e+09   1371.0
80%    7.000017e+09   2573.0
90%    7.000017e+09   4235.0
99%    7.000017e+09  10330.0
max    7.000017e+09  13275.0

В среднем выручка с пользователя 1790р. Разброс от 5 до 13275р

### Recency

Найдем даты последних покупок по пользователю, чтобы описать показатель Recency:

In [10]:
recency = data.groupby('user_id').agg({'date': 'max'}).reset_index().rename(
    columns={'date': 'last_policy_date'})
recency

user_id    last_policy_date
0    7000000090 2023-01-11 08:42:27
1    7000000477 2023-04-27 11:06:31
2    7000000549 2023-08-03 18:42:33
3    7000001437 2023-03-11 10:47:07
4    7000001601 2023-04-11 13:56:56
..          ...                 ...
378  7000017257 2023-09-23 14:57:43
379  7000017261 2023-09-23 10:42:00
380  7000017263 2023-09-22 16:12:43
381  7000017284 2023-09-23 05:52:26
382  7000017301 2023-09-23 12:04:10

[383 rows x 2 columns]

Расчитаем число дней с последней покупки до текущей даты

In [11]:
# до текущей даты
# recency['current_date'] = pd.to_datetime(datetime.date.today())
recency['current_date'] = pd.to_datetime('2023-09-25')
recency['churn_time'] = recency['current_date'] - recency['last_policy_date']
recency

user_id    last_policy_date current_date        churn_time
0    7000000090 2023-01-11 08:42:27   2023-09-25 256 days 15:17:33
1    7000000477 2023-04-27 11:06:31   2023-09-25 150 days 12:53:29
2    7000000549 2023-08-03 18:42:33   2023-09-25  52 days 05:17:27
3    7000001437 2023-03-11 10:47:07   2023-09-25 197 days 13:12:53
4    7000001601 2023-04-11 13:56:56   2023-09-25 166 days 10:03:04
..          ...                 ...          ...               ...
378  7000017257 2023-09-23 14:57:43   2023-09-25   1 days 09:02:17
379  7000017261 2023-09-23 10:42:00   2023-09-25   1 days 13:18:00
380  7000017263 2023-09-22 16:12:43   2023-09-25   2 days 07:47:17
381  7000017284 2023-09-23 05:52:26   2023-09-25   1 days 18:07:34
382  7000017301 2023-09-23 12:04:10   2023-09-25   1 days 11:55:50

[383 rows x 4 columns]

In [12]:
recency.describe(percentiles = [.2, .4, .6, .8, .9, .99])

user_id                   churn_time
count  3.830000e+02                          383
mean   7.000014e+09   56 days 04:26:23.140992167
std    3.321306e+03   62 days 12:50:18.421226199
min    7.000000e+09              0 days 09:22:27
20%    7.000012e+09       8 days 14:57:32.800000
40%    7.000015e+09      22 days 18:36:09.400000
50%    7.000015e+09             33 days 10:46:10
60%    7.000016e+09   48 days 19:24:51.799999999
80%    7.000017e+09   84 days 22:00:35.800000004
90%    7.000017e+09  167 days 09:59:31.200000002
99%    7.000017e+09  253 days 01:02:21.960000004
max    7.000017e+09            264 days 14:48:47

- Среднее время с последней покупки 56 дней
- У 40% пользователей последняя покупка была меньше, чем 22 дня назад
- у 60% пользователей последняя покупка была меньше, чем 48 дней назад
- у 80% пользователей последняя покупка была меньше, чем 84 дня назад
- Минимальное число дней с последней покупки 0, максимальное - 264

### Объдинение таблиц

Объединим таблицы `frequency`, `monetary` и `recency`

In [13]:
rmf = frequency.merge(
    monetary, left_on='user_id', right_on='user_id', how="outer").merge(
    recency[['user_id', 'churn_time']], left_on='user_id', right_on='user_id', how="outer"
)
rmf

user_id  number_policies   amount        churn_time
0    7000000090                1   107.85 256 days 15:17:33
1    7000000477                2   295.50 150 days 12:53:29
2    7000000549                2  1134.26  52 days 05:17:27
3    7000001437                1   314.00 197 days 13:12:53
4    7000001601               25   956.55 166 days 10:03:04
..          ...              ...      ...               ...
378  7000017257                1  1008.75   1 days 09:02:17
379  7000017261                1   971.70   1 days 13:18:00
380  7000017263                2  1381.99   2 days 07:47:17
381  7000017284                1  1327.89   1 days 18:07:34
382  7000017301                1   491.64   1 days 11:55:50

[383 rows x 4 columns]

### Расчет степеней R, M, F

Присвоим степени 1, 2, 3 по каждому свойству

По свойству частоты покупок `Frequency` выберем резделители: больше 1 покупки - 80 процентиль, больше 4 покупок - это 90 процентиль.

По сумме выручки `Monetary` границы степеней выберем: 1083р (чуть больше тысячи) - 50 процентиль, 2573р - 80 процентиль.

По числу дней с последнего полиса `Recency` границы поставим по 33 дням (месяц) - 50 процентиль и 84 дня (3 месяца) - 80 процентиль

In [14]:
# создание столбцов 
rmf[['R', 'M', 'F']] = ''
# степени F
rmf.loc[rmf['number_policies'] > rmf['number_policies'].quantile(0.9), 'F'] = '1'
rmf.loc[rmf['number_policies'] <= rmf['number_policies'].quantile(0.9), 'F'] = '2'
rmf.loc[rmf['number_policies'] <= rmf['number_policies'].quantile(0.8), 'F'] = '3'
# степени M
rmf.loc[rmf['amount'] > rmf['amount'].quantile(0.8), 'M'] = '1'
rmf.loc[rmf['amount'] <= rmf['amount'].quantile(0.8), 'M'] = '2'
rmf.loc[rmf['amount'] <= rmf['amount'].quantile(0.5), 'M'] = '3'
# степени R
rmf.loc[rmf['churn_time'] > rmf['churn_time'].quantile(0.8), 'R'] = '3'
rmf.loc[rmf['churn_time'] <= rmf['churn_time'].quantile(0.8), 'R'] = '2'
rmf.loc[rmf['churn_time'] <= rmf['churn_time'].quantile(0.5), 'R'] = '1'
# вывод таблицы
rmf.head(10)

user_id  number_policies   amount        churn_time  R  M  F
0  7000000090                1   107.85 256 days 15:17:33  3  3  3
1  7000000477                2   295.50 150 days 12:53:29  3  3  3
2  7000000549                2  1134.26  52 days 05:17:27  2  2  3
3  7000001437                1   314.00 197 days 13:12:53  3  3  3
4  7000001601               25   956.55 166 days 10:03:04  3  3  1
5  7000002336                5  1422.00 220 days 07:26:53  3  2  1
6  7000002486                4   422.25 184 days 15:45:19  3  3  1
7  7000002658                7   649.63  58 days 06:12:21  2  3  1
8  7000002803                2    68.30   2 days 17:37:02  1  3  3
9  7000003457                1   762.15  60 days 02:43:26  2  3  3

In [15]:
# оставляем тольку нужные столбцы
rmf = rmf[['user_id', 'R', 'M', 'F']]
rmf

user_id  R  M  F
0    7000000090  3  3  3
1    7000000477  3  3  3
2    7000000549  2  2  3
3    7000001437  3  3  3
4    7000001601  3  3  1
..          ... .. .. ..
378  7000017257  1  3  3
379  7000017261  1  3  3
380  7000017263  1  2  3
381  7000017284  1  2  3
382  7000017301  1  3  3

[383 rows x 4 columns]

## Выделение сегментов

### Сегмент attention

Сегмент attention требует изучения причин оттока, одни из самых крупных клиентов с самыми старыми датами последних заказов, которых мы найдем так:

`M == "1" & R == "3"`

In [16]:
attention = rmf.query('M == "1" & R == "3"').reset_index(drop=True)
attention

user_id  R  M  F
0  7000009123  3  1  3
1  7000010077  3  1  1
2  7000010213  3  1  3
3  7000010219  3  1  3
4  7000011341  3  1  1
5  7000013068  3  1  3
6  7000013191  3  1  3
7  7000013984  3  1  3

Таких пользователей 8. Напишем функию для вывода сегметнов и отобразим сегмент attention:

In [17]:
def view_segment(segment):
    

    current_list = list(segment['user_id'])

    segment_table = data.query('user_id in @current_list').groupby('user_id').agg({'surname': 'max', 
                                 'policy_id': 'count', 'user_source': 'max', 'cost_of_policy': 'sum',
                                 'vehicle': 'max', 'date': 'max'}).reset_index()
    segment_table.columns = ['user_id', 'surname', 'num_policies', 'user_source', 'amount', 
                               'vehicle', 'last_policy_date']
    return segment_table

attention_table = view_segment(attention)
attention_table

user_id       surname  num_policies         user_source    amount  \
0  7000009123        Чернов             1  promocode_sberteam  11805.00   
1  7000010077      Долганов             5                 app   6963.00   
2  7000010213    Гнездилова             2                 app   9790.10   
3  7000010219  Шахимарданов             1      promocode_sber   4094.64   
4  7000011341         Плюст             4      promocode_sber   9165.96   
5  7000013068       Соболев             2      promocode_sber   3744.61   
6  7000013191    Крюковский             1                 app   2599.26   
7  7000013984       Дорохов             2                 app   6628.84   

                      vehicle    last_policy_date  
0  Mercedes-Benz C-Класс 2017 2023-01-03 09:11:13  
1  Mercedes-Benz S-Класс 2019 2023-04-22 12:37:56  
2                Audi A6 2014 2023-03-04 19:19:37  
3        Mitsubishi L200 2022 2023-03-06 13:06:03  
4            Toyota RAV4 2017 2023-06-08 16:07:44  
5      Volkswagen Tiguan 2016 2023-06-09 18:28:00  
6              Acura MDX 2014 2023-06-12 07:44:01  
7         Nissan Qashqai 2018 2023-06-26 11:22:54

Есть пользователи, которые совершали по одной дорогой покупке из-за авто или большого пакета. Или несколько покупок  совокупно на значимую сумму. По сути это все ушедшие клиенты, которые были значимы

### Сегмент лидеров

Или найдем самых классных, верных и упитанных пользователей, чтобы выписать им промокод на полис, свозить на море и прочее:

In [18]:
top = rmf.query('M == "1" & R == "1" & F == "1"').reset_index(drop=True)
top

user_id  R  M  F
0   7000003967  1  1  1
1   7000008023  1  1  1
2   7000009317  1  1  1
3   7000009727  1  1  1
4   7000012165  1  1  1
5   7000013496  1  1  1
6   7000013576  1  1  1
7   7000013909  1  1  1
8   7000014333  1  1  1
9   7000015737  1  1  1
10  7000015798  1  1  1
11  7000016845  1  1  1

Таких пользователей 12, выведем сегмент лидеров

In [19]:
top_table = view_segment(top)
top_table

user_id    surname  num_policies     user_source   amount  \
0   7000003967  Карпенков             7             app  7317.99   
1   7000008023  Бузоверов            39             app  2870.06   
2   7000009317    Ненюков            14  promocode_sber  4336.46   
3   7000009727    Захаров            11             app  3653.91   
4   7000012165    Шейкман            15  promocode_sber  7477.91   
5   7000013496    Чукалин             4        sber_web  9758.72   
6   7000013576    Петрова             4  promocode_sber  8000.91   
7   7000013909  Золотарев             4             app  2678.79   
8   7000014333   Жогличев             4             app  4238.62   
9   7000015737    Петухов             8             app  5153.69   
10  7000015798   Зимняков             8             app  2665.56   
11  7000016845    Любимов             4        sber_web  2895.66   

                               vehicle    last_policy_date  
0                      Skoda Yeti 2018 2023-09-20 00:46:52  
1                          BMW X3 2015 2023-09-23 14:53:43  
2   Land Rover Range Rover Evoque 2011 2023-08-27 12:25:07  
3                   Hyundai Creta 2019 2023-09-14 09:27:43  
4               Mercedes-Benz GLA 2015 2023-08-28 10:23:18  
5                         Audi Q7 2015 2023-08-26 09:34:18  
6               Chery Tiggo 8 Pro 2021 2023-09-01 11:26:36  
7                Hyundai Santa Fe 2018 2023-09-02 04:45:19  
8                      Kia Cerato 2018 2023-09-13 08:31:32  
9                          Kia K5 2020 2023-09-20 10:47:06  
10                        Audi A3 2016 2023-09-13 06:36:02  
11                  Chery Tiggo 8 2020 2023-09-21 08:07:08

Постоянные пользователи, часто покупающие со значимой суммой покупок

- Пользоватеи с наимбольшим числом полисов - это Бузоверов, Ненюков, Шейкман

- Пользователи с наибольшей суммой покупок - это Чукалин, Петрова, Шейкман, Карпенков



### Сегмент потенциала

Последний сегмент, неактивные пользователи, которых нужно вовлекать в дальнейшие покупки, с показателями `M=2/3, F=1, R=1`

In [20]:
potential = rmf.query('M != "1" & R == "1" & F == "1"').reset_index(drop=True)
potential

user_id  R  M  F
0  7000008552  1  3  1
1  7000010004  1  2  1
2  7000014628  1  3  1
3  7000014990  1  3  1
4  7000014994  1  3  1
5  7000015446  1  2  1
6  7000016115  1  3  1
7  7000016212  1  3  1
8  7000016546  1  3  1

Таких пользователей 9, отобразим сегмент потенциала:

In [21]:
potential_table = view_segment(potential)
potential_table

user_id         surname  num_policies     user_source   amount  \
0  7000008552         Смирнов             4             app   518.60   
1  7000010004        Кунчинин            11  promocode_sber  1976.07   
2  7000014628          Буянов             5             app   926.20   
3  7000014990         Новиков            11             app   881.10   
4  7000014994        Костенко             5        sber_web   661.88   
5  7000015446  Мухаметкильдин             5             app  1594.09   
6  7000016115        Смятских             5             app   771.49   
7  7000016212       Ничипурук             4             app   575.89   
8  7000016546         Бондарь             4        sber_web   142.84   

                           vehicle    last_policy_date  
0            Chevrolet Cobalt 2013 2023-09-23 15:33:40  
1            LADA (ВАЗ) Vesta 2019 2023-09-11 10:38:59  
2                  Opel Mokka 2014 2023-09-21 17:13:13  
3  Land Rover Discovery Sport 2020 2023-09-24 09:49:22  
4           Volkswagen Tiguan 2011 2023-08-25 14:16:55  
5           LADA (ВАЗ) Priora 2012 2023-08-29 15:54:41  
6        Land Rover Discovery 2012 2023-09-10 11:26:36  
7            LADA (ВАЗ) Vesta 2018 2023-09-20 13:52:18  
8             LADA (ВАЗ) XRAY 2017 2023-09-19 11:13:43

У этих пользователей выручка меньше, но частота покупок высокая и совершали они их не так давно. Выведем тарифы, которые приобритают этот сегмент

In [28]:
potential_list = list(potential["user_id"])
data.query('user_id in @potential_list').groupby('type').agg({'policy_id': 'count'}).reset_index()

type  policy_id
0    combo          1
1       km         12
2  parking         36
3    theft          3
4     time          2

Этот сегмент основывается на парковках, что логично, частые покупки с низким чеком

## Подготовка презентации

По итогам сегментирования пользователей с помощью анализа RFM подготовлена [презентация](https://github.com/olyautrom/data_analyst_simble/blob/master/1_user_rfm/user_rfm.pdf) 

## Вывод

RFM-анализ – это отличный инструмент для маркетинговой аналитики компании, с помощью которого можно разделить аудиторию и добиться эффективных продаж.

На основании этих списков можно сделать обзвоны, дополнительное вовлечение ушедших и неактивных клиентов, а также анализ и поощрение постоянных.

По итогу RFM-анализу на основе покупок за этот год выделено три сегмента:
- attention
- лидеров
- потенциала

Сегменты зафиксированы в презентации, которая передана продакту и клиентскому сервису для дальнейшей работы